test/e2e/registration_service_test.go

```diff
@@ -429,6 +432,138 @@ func (s *registrationServiceTestSuite) TestSignupOK() {
+
+	// Retrieve the updated UserSignup
+	userSignup, err = s.hostAwait.WaitForUserSignup(identity0.ID.String())
+	require.NoError(s.T(), err)
+
+	// Confirm there is a verification code annotation value, and store it in a variable
+	verificationCode := userSignup.Annotations[v1alpha1.UserSignupVerificationCodeAnnotationKey]
+	require.NotEmpty(s.T(), verificationCode)
+
+	// Confirm the expiry time has been set
+	require.NotEmpty(s.T(), userSignup.Annotations[v1alpha1.UserVerificationExpiryAnnotationKey])
+
+	// Attempt to verify with an incorrect verification code
+	invokeEndpoint(s.T(), "GET", s.route+"/api/v1/signup/verification/invalid", token0, "", http.StatusForbidden)
+
+	// Retrieve the updated UserSignup
+	userSignup, err = s.hostAwait.WaitForUserSignup(identity0.ID.String())
+	require.NoError(s.T(), err)
+
+	// Check attempts has been incremented
+	require.NotEmpty(s.T(), userSignup.Annotations[v1alpha1.UserVerificationAttemptsAnnotationKey])
+
+	// Confirm the verification code has not changed
+	require.Equal(s.T(), verificationCode, userSignup.Annotations[v1alpha1.UserSignupVerificationCodeAnnotationKey])
+
+	// Verify with the correct code
+	invokeEndpoint(s.T(), "GET", s.route + fmt.Sprintf("/api/v1/signup/verification/%s",
+		userSignup.Annotations[v1alpha1.UserSignupVerificationCodeAnnotationKey]), token0, "", http.StatusOK)
+
+	// Retrieve the updated UserSignup
+	userSignup, err = s.hostAwait.WaitForUserSignup(identity0.ID.String())
+	require.NoError(s.T(), err)
+
+	// Confirm all unrequired verification-related annotations have been removed
+	require.Empty(s.T(), userSignup.Annotations[v1alpha1.UserVerificationExpiryAnnotationKey])
+	require.Empty(s.T(), userSignup.Annotations[v1alpha1.UserVerificationAttemptsAnnotationKey])
+	require.Empty(s.T(), userSignup.Annotations[v1alpha1.UserSignupVerificationCodeAnnotationKey])
+	require.Empty(s.T(), userSignup.Annotations[v1alpha1.UserSignupVerificationTimestampAnnotationKey])
+	require.Empty(s.T(), userSignup.Annotations[v1alpha1.UserSignupVerificationCounterAnnotationKey])
+	require.Empty(s.T(), userSignup.Annotations[v1alpha1.UserSignupVerificationInitTimestampAnnotationKey])
+
+	// Call get signup endpoint with a valid token and make sure it's pending approval
+	mp, mpStatus = parseResponse(s.T(), invokeEndpoint(s.T(), "GET", s.route + "/api/v1/signup", token0, "", http.StatusOK))
+	assert.Equal(s.T(), "", mp["compliantUsername"])
+	assert.Equal(s.T(), identity0.Username, mp["username"])
+	require.IsType(s.T(), false, mpStatus["ready"])
+	assert.False(s.T(), mpStatus["ready"].(bool))
+	assert.Equal(s.T(), "PendingApproval", mpStatus["reason"])
+	require.False(s.T(), mpStatus["verificationRequired"].(bool))
+
+	// Now approve the usersignup.
+	userSignup.Spec.Approved = true
+	err = s.hostAwait.Client.Update(context.TODO(), userSignup)
+	require.NoError(s.T(), err)
+
+	// Confirm the MasterUserRecord is provisioned
+	_, err = s.hostAwait.WaitForMasterUserRecord(identity0.Username)
```

Comment:
if we want to check that it's provisioned then we should verify the existence of the condition:
```suggestion
	_, err = s.hostAwait.WaitForMasterUserRecord(identity0.Username, wait.UntilMasterUserRecordHasCondition(Provisioned()))
```

Reply:
Good idea, updated.